services/twitter.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-console */
+
+import logger from "./logger";
+import testimonialsData from "../data/testimonials.json";
+import tweetFixtures from "../fixtures/twitterTweetsApiFixture.json";
+
+const API_URL =
+  "/twitter/2/tweets?user.fields=profile_image_url,username&expansions=author_id&ids=";
+const DEV_MODE = "DEV_MODE";
+
+class Twitter {
+
+  constructor(axios) {
+    this.axios = axios;
+  }
+
+  async _fetch() {
+    const token = this._getToken();
+    const options = {
+      headers: { Authorization: `Bearer ${token}` },
+      mode: "no-cors",
+    };
+
+    let json = "";
+    if (token === DEV_MODE) {
+      logger.info(
+        "Twitter instantiated using dev mode; sample tweets will be loaded from fixtures."
+      );
+      json = tweetFixtures;
+    } else {
+      try {
+        const res = await this.axios.get(
+          `${API_URL}${testimonialsData.tweets}`,
+          options
+        );
+        json = res.data;
+      } catch (e) {
+        logger.error("Could not retrieve tweets from Twitter API.", e);
+      }
+    }
+
+    return json;
+  }
+
+  _getToken() {
+    return process.env.TWITTER_API_BEARER_TOKEN || DEV_MODE;
+  }
+
+  _createUserMap(apiUserData) {
+    /* Maps users by id. */
+    const userMap = {};
+    for (const user of apiUserData) {
+      userMap[user.id] = user;
+    }
+    return userMap;
+  }
+
+  _annotateTweetsWithUsers(apiTweetData, userMap) {
+    /* The twitter API separates user information and tweet content. This method recombines, overwriting fields as necessary. */
+    let annotated = [];
+    for (const tweet of apiTweetData) {
+      const annotatedTweet = Object.assign(tweet, {
+        name: userMap[tweet.author_id].name,
+        username: userMap[tweet.author_id].username,
+        profile_image_url: userMap[tweet.author_id].profile_image_url,
+        profile_image_url_original: userMap[
+          tweet.author_id
+        ].profile_image_url.replace(/_normal\./, "."),
+      });
+      annotated.push(annotatedTweet);
+    }
+    return annotated;
+  }
+
+  async get() {
+    const tweets = await this._fetch();
+    const userMap = this._createUserMap(tweets.includes.users);
```

Comment:
should validate the `tweets` object since it's coming from an external source.

Reply:
What do we think the behavior should be on a failed fetch?

Should we
a) Load fixtures, since this isn't mission critical
or
b) Render empty widget
or
c) Fail render
?

Reply:
hmmmmmmm...fixtures make sense to me since it will still provide a nice experience

Reply:
Done. I un-prefixed the 'TEST' I had in the beginning of the fixture strings now that there's a chance we might use them on the prod page. I'd thought about this before too; just hadn't done it.

Reply:
we also have a `data` folder we should move it to instead of testfixtures, since it's prod data.  (at least if we plan to keep two separate folders for static data)

Reply:
Yeah, was debating on that. I could go either way on it, as it really is intended to only be fallback and we don't want to give the impression that this is where we canonically want to load from, but otoh, we now have the possibility of using this if things go south.